clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/deploy/ops/CloneKubernetesAtomicOperation.groovy

```diff
@@ -0,0 +1,109 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.clouddriver.kubernetes.deploy.ops
+
+import com.netflix.frigga.Names
+import com.netflix.spinnaker.clouddriver.data.task.Task
+import com.netflix.spinnaker.clouddriver.data.task.TaskRepository
+import com.netflix.spinnaker.clouddriver.kubernetes.deploy.KubernetesUtil
+import com.netflix.spinnaker.clouddriver.kubernetes.deploy.description.CloneKubernetesAtomicOperationDescription
+import com.netflix.spinnaker.clouddriver.kubernetes.deploy.description.KubernetesContainerDescription
+import com.netflix.spinnaker.clouddriver.kubernetes.deploy.description.KubernetesResourceDescription
+import com.netflix.spinnaker.clouddriver.orchestration.AtomicOperation
+import io.fabric8.kubernetes.api.model.ReplicationController
+import org.springframework.beans.factory.annotation.Autowired
+
+class CloneKubernetesAtomicOperation implements AtomicOperation<Void> {
+  private static final String BASE_PHASE = "CLONE_SERVER_GROUP"
+
+  @Autowired
+  KubernetesUtil kubernetesUtil
+
+  CloneKubernetesAtomicOperation(CloneKubernetesAtomicOperationDescription description) {
+    this.description = description
+  }
+
+  private static Task getTask() {
+    TaskRepository.threadLocalTask.get()
+  }
+
+  CloneKubernetesAtomicOperationDescription description
+
+/*
+ * curl -X POST -H "Content-Type: application/json" -d  '[ { "cloneServerGroup": { "source": { "serverGroupName": "k8s-test-v000" }, "credentials":  "my-k8s-account" } } ]' localhost:7002/kubernetes/ops
+ * curl -X POST -H "Content-Type: application/json" -d  '[ { "cloneServerGroup": { "stack": "prod", "freeFormDetails": "mdservice", "targetSize": "4", "source": { "serverGroupName": "k8s-test-v000" }, "credentials":  "my-k8s-account" } } ]' localhost:7002/kubernetes/ops
+*/
+  // TODO: should return DeploymentResult object
+  @Override
+  Void operate(List priorOutputs) {
+    CloneKubernetesAtomicOperationDescription newDescription = cloneAndOverrideDescription()
+
+    task.updateStatus BASE_PHASE, "Initializing copy of server group for " +
+      "${description.application}-${description.stack}..."
+
+    new DeployKubernetesAtomicOperation(newDescription).operate(priorOutputs)
+
+    task.updateStatus BASE_PHASE, "Finished copying server group for " +
+      "${description.application}-${description.stack}. "
+  }
+
+  // Made public for testability, not intended to be called directly
```

Comment:
Comment isn't really necessary - everything in groovy is more-or-less public and `private` modifier are more for documentation purposes. See http://www.gmarwaha.com/blog/2009/02/02/groovy-private-is-not-private/